test/tests/mortar/3d-periodic/tests

```diff
@@ -6,6 +6,7 @@
     input = 'periodic.i'
     exodiff = 'periodic_out.e'
     requirement = 'We shall be able to enforce periodic boundary conditions on 3d domains using mortar constraints.'
+    abs_zero = 1e-08
```

Comment:
@lindsayad, do you know what's the physical meaning of `lm`s in this problem? The changes I am making here affect the converged value of the Lagrange multipliers by a couple of orders of magnitude, but the primal solution passes just fine.

Reply:
It represents the diffusive flux. If the solution changes a lot, that's a little concerning

Reply:
That seems like it's in the noise to me. I would wager that perhaps the nonlinear solve histories are a little bit different? Maybe one is more converged than the other?

Reply:
This problem converges pretty tightly. The residual comes down 15 orders of magnitude. 

Making the tolerance less tight does change the solution a bit (20%) but not like that large change from changing the projection tolerance. Interestingly, the change in solver tolerance from say 1e-14 to 1e-10 that causes a 20% change in LMs doesn't seem to affect the primal solution (the first five digits stay the same).

Reply:
What are the largest fluxes in that problem?

Reply:
That's what I don't fully understand, the largest fluxes are "Before ~1.0e-13 after ~1.0e-10." ballpark. Below is an image of how this looks like after the projection tolerance change:

<img width="938" alt="image" src="https://user-images.githubusercontent.com/14300313/192850920-47242261-dda2-45fb-9401-cd28ceba7f73.png">

Reply:
Oh ... yea that seems pretty wack ... it's as if there is no diffusive flux. Maybe this is a poorly setup test. Is the primal solution constant everywhere?

Reply:
Okay, I think I know. If I remove the equal value constraint, the primal solution is identical (not constant, it's 7.2 at the center of the cylinder and 1.0 on the outer surface).

Reply:
Ok so it seems like the constraint is not doing anything? I think @jbadger95 created this test. Maybe he has some thoughts

Reply:
Yeah, it seems as if the equal value condition is fulfilled without the constraint. I think the test was first used as "cant-do-3d" example, which may explain the setup. I can work on making updates to it.

Reply:
Yea if we can make it so that the "null" LM solution is a solution, that would be awesome

Reply:
You marked this as resolved ... does that mean you have a non-trival LM solution now with that body force change?

Reply:
Yeah. The problem now, without the mortar constraint, doesn't have the same solution on the boundaries on which the  constraint is acting. The LMs are on the order of 1.

Reply:
nice!

Reply:
Hi, sorry for the late reply, I'm traveling till Oct. 9. 

It's been a while but I think there was a tetrahedron version of this test that failed when refining (there was this oddity about how generating a tet mesh from a hexahedral mesh, then refining was unstable; when you just generate a new tet mesh from a refined hexahedral mesh things were fine). I think I was using this geometry because it was a bit less regular than the cubes a lot of my other tests were defined on. Even thought the flux was supposed to be 0, this instability would pop up for refined tet meshes and I think I had to add stabilization or something. Anyway I'm not sure if this info helps but I think those stability issues were the reason for the test initially; probably not a bad idea to have a nontrivial LM solution though!

Sorry to hear the projection is throwing an error! @recuero, I tried to give tolerances the same dimensions as the quantities being toleranced; we ran into a problem when length-scale of the problem was small in the units being used and so a bunch of stuff was being toleranced prematurely. Perhaps that non-dimensionalization is dropping the tolerance below machine precision?

Reply:
Thanks, Jacob, for chiming in. 

We found that projections were unable to hit the projection tolerance for elements with lengths on the order of 10 or 20. After normalizing the tolerance, we can run that problem, so, yeah, my guess is that always using the hard-coded value was causing demands below machine precision.

I tried to generalized the "fix" for small displacement values (i.e. making the tolerance tighter in those cases) and got a bunch of failures with non-converged projections. It appears we are walking a fine line.